doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1737,33 +1777,79 @@ Controlling the proof flow
    impossible. In most cases, :g:`P` or :g:`∼P` is one of the hypotheses of the
    local context.
 
-.. tacn:: contradiction
-   :name: contradiction
+.. tacn:: contradiction {? @one_term_with_bindings }
 
-   This tactic applies to any goal. The contradiction tactic attempts to
-   find in the current context (after all intros) a hypothesis that is
-   equivalent to an empty inductive type (e.g. :g:`False`), to the negation of
-   a singleton inductive type (e.g. :g:`True` or :g:`x=x`), or two contradictory
-   hypotheses :g:`P` and :g:`not P`.
+   Tries to prove the current goal by finding a contradiction.
 
-   .. exn:: No such assumption.
-      :undocumented:
+   If :n:`@one_term_with_bindings` is not provided (the most common use case),
+   the tactic first does an :tacn:`intros`.  The tactic then proves the goal if
+
+   - the updated context has a pair of hypotheses where one is the negation of
+     the other (e.g. :n:`P` and not :n:`~P`), or
+   - there is a hypothesis with an empty inductive hypothesis (e.g. :n:`False`), or
+   - there is a hypothesis that is the negation of a singleton inductive type
+     (e.g. :n:`True` or :n:`x=x`).
+
+   If :n:`@one_term_with_bindings` is provided, the type of
+   :n:`@one_term_with_bindings` must be a negation, such as :n:`~P`,
+   or :n:`False` (or another empty inductive type).
```

Comment:
While this sentence is correct, every time I read it, I struggle. I propose to change it like the following, as I think this would make it more symmetric and easier to read:
```suggestion
   If :n:`@one_term_with_bindings` is provided, the type of
   :n:`@one_term_with_bindings` must be either a negation, such as :n:`~P`,
   or an empty inductive type, such as :n:`False`.
```